finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Row.scala

```diff
@@ -130,14 +135,19 @@ class BinaryEncodedRow(rawRow: Buf, val fields: IndexedSeq[Field], indexMap: Map
     for ((field, idx) <- fields.zipWithIndex) yield {
       if (isNull(idx)) NullValue
       else field.fieldType match {
-        case Type.Tiny        => ByteValue(reader.readByte())
-        case Type.Short       => ShortValue(reader.readShortLE())
-        case Type.Int24       => IntValue(reader.readIntLE())
-        case Type.Long        => IntValue(reader.readIntLE())
-        case Type.LongLong    => LongValue(reader.readLongLE())
-        case Type.Float       => FloatValue(reader.readFloatLE())
-        case Type.Double      => DoubleValue(reader.readDoubleLE())
-        case Type.Year        => ShortValue(reader.readShortLE())
+        case Type.Tiny if field.isSigned      => ByteValue(reader.readByte())
+        case Type.Tiny                        => ShortValue(reader.readUnsignedByte())
+        case Type.Short if field.isSigned     => ShortValue(reader.readShortLE())
+        case Type.Short                       => IntValue(reader.readUnsignedShortLE())
+        case Type.Int24 if field.isSigned     => IntValue(reader.readIntLE())
+        case Type.Int24                       => IntValue(reader.readIntLE())
```

Comment:
Fun fact: this has been broken for a while: we should not be reading a 4 byte int for the Int24 type, but only 3 https://dev.mysql.com/doc/refman/5.7/en/integer-types.html. This isn't a problem with this PR per say, just what we've been doing wrong for a while now.